glusterd2/xlator/options/options.go

```diff
@@ -260,16 +260,28 @@ func ValidatePriorityOrSize(o *Option, val string) error {
 
 // ValidateSizeList validates if the option is a valid size list
 func ValidateSizeList(o *Option, val string) error {
+	var sizeinbytes int64
 	slist := strings.Split(val, ",")
 	for _, el := range slist {
 		l := len(el)
 		if strings.ContainsRune(el, 'B') || strings.ContainsRune(el, 'b') {
+			size := el[l-2 : l]
 			t := strings.TrimSpace(el[:l-2])
 			v, err := strconv.ParseInt(t, 10, 64)
 			if err != nil {
 				return err
-			} else if v%512 != 0 {
-				return ErrInvalidArg
+			} else {
+				switch size {
+				case "KB", "kb":
+					sizeinbytes = v * 1000
```

Comment:
It should be 1000 or 1024?

Reply:
As discussed, I think 1KB=1000bytes and 1KiB=1024bytes.